src/smartcontracts/argSerializer.ts

```diff
@@ -74,13 +74,22 @@ export class ArgSerializer {
                 let typedValue = readValue(type.getFirstTypeParameter());
                 return new OptionalValue(type, typedValue);
             } else if (type.hasExactClass(VariadicType.ClassName)) {
+                let variadicType = <VariadicType>type;
                 let typedValues = [];
 
-                while (!hasReachedTheEnd()) {
-                    typedValues.push(readValue(type.getFirstTypeParameter()));
+                if (variadicType.isCounted) {
+                    const count: number = readValue(new U32Type()).valueOf().toNumber();
+
+                    for (let i = 0; i < count; i++) {
+                        typedValues.push(readValue(type.getFirstTypeParameter()));
+                    }
+                } else {
+                    while (!hasReachedTheEnd()) {
```

Comment:
Non-counted variadic arguments (regular variadic arguments) can only be at the tail of the arguments.